frontend/src/modules/report/FeatureListContainer/FeatureListContainer.jsx

```diff
@@ -39,211 +40,194 @@ class FeatureListContainer extends Component {
         skipped: true,
       },
       expandedTagsList: [],
+      loading: false,
     };
-    this.handleViewSwitch = this.handleViewSwitch.bind(this);
-    this.handleTagsSwitch = this.handleTagsSwitch.bind(this);
-    this.handleEditModeSwitch = this.handleEditModeSwitch.bind(this);
-    this.handleFilterButtonClick = this.handleFilterButtonClick.bind(this);
-    this.handleTagSelect = this.handleTagSelect.bind(this);
-    this.handleTagAssigned = this.handleTagAssigned.bind(this);
-    this.handleWarningShow = this.handleWarningShow.bind(this);
-    this.handleWarningClosed = this.handleWarningClosed.bind(this);
-    this.handleTagIgnore = this.handleTagIgnore.bind(this);
   }
 
   componentDidMount() {
     const featureList = new FeatureList();
     const { product, version, build } = this.props;
+    this.setState({ loading: true });
     Promise.all([
       getFeatureListByTagData(product, version, build),
       getSimpleFeatureListData(product, version, build),
       getTagAssignmentData(product, version, build),
       getIgnoredTags(product),
-    ]).then(data => {
-      featureList.setFeatureListByTag(data[0]);
-      featureList.setSimpleFeatureList(data[1]);
-      featureList.setUserFromTagAssignments(data[2]);
-      featureList.setIgnoredTags(data[3]);
-      this.setState({ featureList });
-    });
-  }
-
-  handleViewSwitch() {
-    this.setState(prevState =>
-      Object.assign({}, prevState, {
-        isTagView: !prevState.isTagView,
+    ])
+      .then(data => {
+        featureList.setFeatureListByTag(data[0]);
+        featureList.setSimpleFeatureList(data[1]);
+        featureList.setUserFromTagAssignments(data[2]);
+        featureList.setIgnoredTags(data[3]);
+        this.setState({ featureList });
       })
-    );
+      .finally(() => this.setState({ loading: false }));
```

Comment:
I'm guessing all of the data fetching logic should ideally move out of this component and into services, which would mean that this loading flag wouldn't be a local state anymore and would move into the redux store?

Reply:
Yeah 100%, I was figuring once we move the bulk of the loaded data to state we can have a single loading component at the top level and we update the loading state in store directly